File: OneDrive/Desktop/COEXIST/FrontCoexist-final1Angular/FrontCoexist-final1/src/app/Services/event.service.ts
import { Injectable } from '@angular/core';
import {HttpClient, HttpParams} from "@angular/common/http";
import {catchError, map, Observable, tap, throwError} from "rxjs";
import { Event } from '../entity/Event';
import {LikedEvents} from "../entity/LikedEvents";


@Injectable({
  providedIn: 'root'
})
export class EventService {
  private apiUrl = 'http://localhost:8000';
  imageBaseUrl = 'http://localhost:8000/'

  constructor(private http: HttpClient) {}

  addEvent(eventData: FormData): Observable<Event> {
    return this.http.post<Event>(`${this.apiUrl}/add-event`, eventData);
  }

  shareEventOnFacebook(eventId: number): Observable<string> {
    return this.http.post<string>(`${this.apiUrl}/${eventId}/share-fb`, null);
  }





  updateEvent(event: Event, eventId: number): Observable<Event> {
    return this.http.put<Event>(`${this.apiUrl}/updateEvent/${eventId}`, event);
  }


  getEvent(eventId: number): Observable<Event> {
    return this.http.get<Event>(`${this.apiUrl}/findByIdEvent/${eventId}`);
  }


  getAllEvents(): Observable<Event[]> {
    return this.http.get<Event[]>(`${this.apiUrl}/retrieveAllEvents`);
  }





  searchEventsByDate(eventDate: string): Observable<Event[]> {
    return this.http.get<Event[]>(`${this.apiUrl}/search/date?eventDate=${eventDate}`);
  }

  searchEventsByDescription(eventDescription: string): Observable<Event[]> {
    return this.http.get<Event[]>(`${this.apiUrl}/search/description?eventDescription=${eventDescription}`);
  }

  searchEventsByLocation(eventLocation: string): Observable<Event[]> {
    return this.http.get<Event[]>(`${this.apiUrl}/search/location?eventLocation=${eventLocation}`);
  }







  getImageUrl(imagePath: string): string {
    return this.imageBaseUrl + imagePath;
  }



  addOrUpdateReaction(eventId: number, userId: number, isLiked: boolean): Observable<any> {
    const url = `${this.apiUrl}/events/${eventId}/reaction`;

    const params = new HttpParams()
      .set('userId', userId.toString())
      .set('eventId', eventId.toString())
      .set('isLiked', isLiked.toString()); // Convert boolean to string

    return this.http.put<any>(url, {}, { params });
  }


  deleteEvent(eventId: number): Observable<void> {
    return this.http.delete<void>(`${this.apiUrl}/deleteEventById/${eventId}`);
  }

  searchEvents(keyword: string): Observable<Event[]> {
    return this.http.get<Event[]>(`${this.apiUrl}/recherche/${keyword}`);
  }
}

File: OneDrive/Desktop/COEXIST/backcoexist-final1SpringBoot/src/main/java/tn/esprit/coexist/service/EventService/EventService.java
package tn.esprit.coexist.service.EventService;

import org.springframework.web.multipart.MultipartFile;
import tn.esprit.coexist.entity.Event;
import tn.esprit.coexist.entity.User;

import java.time.LocalDateTime;
import java.util.Date;
import java.util.List;

public interface EventService {



    public User addUser(User user);
    public Event AddEvent(Event event, MultipartFile imageFile);


    public Event updateEvent(Event event, Integer eventId);
    public void addOrUpdateReaction(Event event, Integer userId, Boolean isLiked, Date likedAt);
    public String shareFb(Integer eventId);


    public Event findById(Integer eventId);

    public List<Event> retrieveAllEvents();
    Event findMostLikedEvent();
    public User findUserById(Integer UserId);





    public void deleteEventById(Integer eventId);


    public List<Event> recherche(String keyword);


    List<Event> searchByDate(LocalDateTime eventDate);

    List<Event> searchByDescription(String eventDescription);

    List<Event> searchByLocation(String eventLocation);




}